### PR TITLE
Bump max Rails version to 7.2

### PR DESCRIPTION
**Pull request by eliotjordan**
Bump max Rails version to 7.2 in Blacklight 7.x. Allow implementers time to upgrade from older versions of Rails that will be EOL soon.

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -131,6 +131,26 @@ jobs:
       run: bundle exec rake ci
       env:
         ENGINE_CART_RAILS_OPTIONS: '--skip-git --skip-keeps --skip-action-cable --skip-test'
+  test_rails7_2:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby: [3.1, 3.2]
+    env:
+      RAILS_VERSION: 7.1.3.4
```
Comment (jcoyne) [inline .github/workflows/ruby.yml:140]: Shouldn't this be 7.2.0?

**Review by jcoyne — CHANGES_REQUESTED** [review]

**Comment by eliotjordan** [comment]
@jcoyne Fixed.

File: .github/workflows/ruby.yml [hunk 2]
```diff
@@ -131,6 +131,26 @@ jobs:
...
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby }}
+        bundler: 'default'
+    - name: Install dependencies
+      run: bundle install
+    - name: Run tests
+      run: bundle exec rake ci
+      env:
+        ENGINE_CART_RAILS_OPTIONS: '--skip-spring --skip-keeps --skip-test'
```
Comment (jcoyne) [inline .github/workflows/ruby.yml:153]:
```suggestion
        ENGINE_CART_RAILS_OPTIONS: '--skip-keeps --skip-test'
```

`--skip-spring ` is not an option in 7.2

**Comment by eliotjordan** [comment]
@jcoyne This should be read to go.

**Comment by jcoyne** [comment]
@eliotjordan it looks like we need to merge https://github.com/projectblacklight/blacklight/pull/3271 first.

**Comment by eliotjordan** [comment]
@jcoyne Trey merged https://github.com/projectblacklight/blacklight/pull/3271. I rebased the changes into this branch.

**Review by jcoyne — APPROVED** [review]

**Comment by jrochkind** [comment]
thank you!

**Pull request merged**